src/orquestra/sdk/_base/_dsl.py

```diff
@@ -625,6 +625,9 @@ def __iter__(self):
         ]
         return iter(futures)
 
+    def __reduce_ex__(self, protocol):
+        raise NotImplementedError("ArtifactFuture cannot be pickled")
```

Comment:
**question**: what happens if this method isn't defined at all?

Reply:
If this isn't defined, `ArtifactFuture` objects can be pickled. This would happen when an artifact is inside a container, such as a `dict` or `list`.

We currently don't support this, so we need to prevent this from happening. With `dill`, it failed to pickle something else in this instance so it kinda worked by accident.

Reply:
**non-blocking**: please state that in a comment or docstring

Reply:
done in 162fd65